docs/locations-and-scheduling.md

```diff
@@ -29,77 +29,106 @@ The APIs used for Compute as a Service are:
-  start the process of removing the namespace from the sync target.
-- `Unbound` – the namespace has been removed and with that released by the sync target and with that from a syncer.
-  This state is set by the namespace controller. The placement controller will notice, and remove the entry in the placement annotation.
+```yaml
+apiVersion: scheduling.kcp.dev/v1alpha1
+kind: Placement
+metadata:
+  name: aws
+spec:
+  locationSelectors:
+  - matchLabels:
+      cloud: aws
+  namespaceSelector:
+    matchLabels:
+      app: foo
+  locationWorkspace: root:default:location-ws
+```
 
-The location strings are of the form `<locationClusterName>+<locationName>+<syncTargetIdentifier>`.
+A matched location will be selected for this `Placement` at first, which makes the `Placement` turns from `Pending` to `Unbound`. Then if there is at
+least one matched ns, the ns will be annotated with `scheduling.kcp.dev/placement` and the placement turns from `Unbound` to `Bound`. After this, a
+`SyncTarget` will be selected from the location picked by the placement.  `state.internal.workload.kcp.dev/<cluster-id>` label with value of `Sync` will
+be set if a valid `SyncTarget` is selected.
 
-Example:
+The use can create another placement targeted to a different location for this ns, e.g. 
 
 ```yaml
-apiVersion: v1
-kind: Namespace
-  name: default
-  annotations:
-    scheduling.kcp.dev/placement: {"root:compute+us-east1-gcp+a7fcajg8a-a9sf-a738":"Bound"}
+apiVersion: scheduling.kcp.dev/v1alpha1
+kind: Placement
+metadata:
+  name: gce
+spec:
+  locationSelectors:
+  - matchLabels:
+      cloud: gce
+  namespaceSelector:
+    matchLabels:
+      app: foo
+  locationWorkspace: root:default:location-ws
 ```
 
-Note: multiple sync targets can be bound at the same time.
+which wiil finally result in another `state.internal.workload.kcp.dev/<cluster-id>` label added to the ns, and the ns will have two different
+`state.internal.workload.kcp.dev/<cluster-id>` label.
 
-The placement state machine is (to be) protected against mutation by the user via admission. The user interface
-to influence the placement decisions is (will be) the `Placement` object.
+Placement is in the `Ready` status condition when
 
-### Resource Syncing
+1. selected location matchs the `Placement` spec.
+2. selected location exists in the location workspace.
 
-As soon as the `scheduling.kcp.dev/placement` annotation is set with state `Pending` on a namespace, the workload namespace 
-controller will pick up the namespace and
+#### Sync target removing
 
-1. set the `scheduling.kcp.dev/placement` annotation state to `Bound` and
-2. set the `state.internal.workload.kcp.dev/<cluster-id>` label to `Sync`.
+A sync target will be removed when:
+
+1. corresponding `Placement` is deleted.
+2. corresponding `Placement` is not in `Ready` condition.
+3. corresponding `SyncTarget` is evicting/not Ready/deleted
```

Comment:
am curious: do we really evict on "not ready" of the sync target? We have to rethink this topic. In kube, I believe there is some larger timeout (10min?) of unreadiness. Here we talk about clusters. I think we should have an even longer grace period, maybe even infinite if the user wants it, i.e. this must likely become user configurable.

It's also related to https://kubernetes.io/blog/2022/05/20/kubernetes-1-24-non-graceful-node-shutdown-alpha/.

Reply:
we currently do this. We could introduce a taint/toleration for clusters something like this? https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/20-taint-toleration

Reply:
We probably want taint/toleration eventually. But I don't think it has to do with the eviction topic, has it? Do we currently schedule away from invalid clusters? We definitely need more controller for that.

Reply:
Can you create an issue to come back to this? I tend to even see this as a time-bomb. I think it's far too aggressive and could be fatal if a cluster blibs for reasons like cloud provider hickups.